src/k8s/pkg/utils/control.go

```diff
@@ -0,0 +1,36 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+// WaitUntilReady waits until the specified condition becomes true.
+func WaitUntilReady(ctx context.Context, checkFunc func() bool, timeout time.Duration, errorMessage string) error {
+	ch := make(chan struct{}, 1)
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				ready := checkFunc()
+				if ready {
+					ch <- struct{}{}
+					return
+				}
+			}
+			time.Sleep(time.Second)
+		}
```

Comment:
You can avoid the extra goroutine and double context cancellation checks by using a `<-time.After(time.Second)` case for the retries instead